src/render/gui/main_menu.cpp

```diff
@@ -0,0 +1,299 @@
+    ImGui::TextUnformatted(error_msg);
+}
+
+// ======================================================================
+
+
+static void ChangeGameState(ThreadedLoopCommon& common, const ThreadedLoopCommon::GameState new_state) {
+    SetVisible(render::Menu::MainMenu, false);
+
+    common.mainMenuData.currentMenu = MainMenuData::Window::main;
+    common.gameState                = new_state;
+}
+
+static void NewGameMenu(ThreadedLoopCommon& common) {
+    const render::GuiMenu menu;
+    render::SetupNextWindowCenter({GetMainMenuWidth(), GetMainMenuHeight()});
+    menu.Begin("_new_game_menu");
+
+    const render::GuiTitle title;
+    title.Begin("New game");
+
+
+    auto seed = common.GetDataGlobal().worlds[0].GetWorldGeneratorSeed(); // Should be same for all worlds
+    ImGui::InputInt("Seed", &seed);
+
+
+    MenuBackButton(common.mainMenuData, MainMenuData::Window::main);
+    SameLineMenuButtonMini(2);
+
+    if (MenuButtonMini("Start")) {
+        data::PrepareWorldDataClear(common.gameDataLocal, common.GetDataGlobal());
+
+        common.ResetGlobalData();
+
+        ChangeGameState(common, ThreadedLoopCommon::GameState::in_world);
+    }
+
+
+    // Sets seed for newly reset global data as well
+    for (auto& world : common.GetDataGlobal().worlds) {
+        world.SetWorldGeneratorSeed(seed);
+    }
+}
+
+///
+/// Lists all save games, loads save game user clicks on
+void LoadSaveGameMenu(ThreadedLoopCommon& common) {
+    using namespace render;
+
+    const GuiMenu menu;
+    SetupNextWindowCenter({GetMainMenuWidth(), GetMainMenuHeight()});
+    menu.Begin("_save_game_browser_menu");
+
+    const GuiTitle title;
+    title.Begin("Load game");
+
+    auto& last_load_error = common.mainMenuData.lastLoadError;
+
+    for (const auto& save_game : data::GetSaveDirIt()) {
+        const auto filename = save_game.path().stem().string();
+
+        if (MenuButton(filename.c_str())) {
+            try {
+                data::DeserializeGameData(common.gameDataLocal, common.GetDataGlobal(), filename);
+            }
+            catch (cereal::Exception& e) {
+                last_load_error = e.what();
+                LOG_MESSAGE_F(error, "Failed to load game %s : %s", filename.c_str(), e.what());
+            }
+        }
+    }
+
+    MenuBackButton(common.mainMenuData, MainMenuData::Window::main);
+    SameLineMenuButtonMini(2);
+
+    if (MenuButtonMini("Play")) {
+        ChangeGameState(common, ThreadedLoopCommon::GameState::in_world);
+    }
+}
+
+///
+/// Asks for save name and saves current world
+void SaveGameMenu(ThreadedLoopCommon& common) {
+    using namespace render;
+
+    const GuiMenu menu;
+    SetupNextWindowCenter({GetMainMenuWidth(), GetMainMenuHeight()});
+    menu.Begin("_save_game_menu");
+
+    const GuiTitle title;
+    title.Begin("Save game");
+
+
+    auto* save_name       = common.mainMenuData.saveName;
+    auto& last_save_error = common.mainMenuData.lastSaveError;
```

Comment:
Add the ability to show the error message to the usage.

When the user acknowledges the error, last_save_error can be cleared